mentat/edit_history.py

```diff
@@ -0,0 +1,112 @@
+import os
+from pathlib import Path
+
+import attr
+from termcolor import colored
+
+
+# All paths should be abs paths
+@attr.define()
+class RenameAction:
+    old_file_name: Path = attr.field()
+    cur_file_name: Path = attr.field()
+
+    def undo(self) -> str | None:
+        if self.old_file_name.exists():
+            return colored(
+                f"File {self.old_file_name} already exists; unable to undo rename from"
+                f" {self.cur_file_name}",
+                color="light_red",
+            )
+        else:
+            os.rename(self.cur_file_name, self.old_file_name)
+
+
+@attr.define()
+class CreationAction:
```

Comment:
Maybe these can be `CreateAction` / `DeleteAction` to match `CreateFile`/`DeleteFile`